code/modules/mob/living/carbon/xenomorph/castes/warlock/abilities_warlock.dm

```diff
@@ -0,0 +1,598 @@
+		if(WEST)
+			lower_left = locate(owner.x - 2, owner.y - 1, owner.z)
+			upper_right = locate(owner.x - 1, owner.y + 1, owner.z)
+		if(EAST)
+			lower_left = locate(owner.x + 1, owner.y - 1, owner.z)
+			upper_right = locate(owner.x + 2, owner.y + 1, owner.z)
+
+	for(var/turf/affected_tile AS in block(lower_left, upper_right)) //everything in the 2x3 block is found.
+		affected_tile.Shake(4, 4, 2 SECONDS)
+		for(var/atom/movable/affected in affected_tile)
+			if(!ishuman(affected) && !istype(affected, /obj/item) && !isdroid(affected))
+				affected.Shake(4, 4, 20)
+				continue
+			if(ishuman(affected))
+				var/mob/living/carbon/human/H = affected
+				if(H.stat == DEAD)
+					continue
+				H.apply_effects(0.5, 0.5)
+				shake_camera(H, 2, 1)
+			var/throwlocation = affected.loc
+			for(var/x in 1 to 6)
+				throwlocation = get_step(throwlocation, active_shield.dir)
+			affected.throw_at(throwlocation, 4, 1, owner, TRUE)
+
+	playsound(owner,'sound/effects/bamf.ogg', 75, TRUE)
+	playsound(owner, "alien_roar", 50)
+
+	GLOB.round_statistics.psy_shield_blasts++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "psy_shield_blasts")
+
+
+/obj/effect/xeno/shield
+	icon = 'icons/Xeno/96x96.dmi'
+	icon_state = "shield"
+	resistance_flags = BANISH_IMMUNE|UNACIDABLE|PLASMACUTTER_IMMUNE
+	max_integrity = 350
+	layer = ABOVE_MOB_LAYER
+	///Who created the shield
+	var/mob/living/carbon/xenomorph/owner
+	///All the projectiles currently frozen by this obj
+	var/list/frozen_projectiles = list()
+
+/obj/effect/xeno/shield/Initialize(loc, creator)
+	. = ..()
+	owner = creator
+	if(!owner)
+		return
+	dir = owner.dir
+	max_integrity = owner.xeno_caste.shield_strength
+	obj_integrity = max_integrity
+	if(dir & (EAST|WEST))
+		bound_height = 96
+		bound_y = -32
+		pixel_y = -32
+	else
+		bound_width = 96
+		bound_x = -32
+		pixel_x = -32
+
+/obj/effect/xeno/shield/projectile_hit(obj/projectile/proj, cardinal_move, uncrossing)
+	if(!(cardinal_move & REVERSE_DIR(dir)))
+		if(!uncrossing)
+			proj.uncross_scheduled += src
+		return FALSE
+	if(uncrossing)
+		return FALSE
+
+	return TRUE
+
+/obj/effect/xeno/shield/do_projectile_hit(obj/projectile/proj)
+	proj.flags_projectile_behavior |= PROJECTILE_FROZEN
+	proj.iff_signal = null
+	frozen_projectiles += proj
+	take_damage(proj.damage, proj.ammo.damage_type, proj.ammo.armor_type, 0, turn(proj.dir, 180), proj.ammo.penetration)
+	if(obj_integrity <= 0)
+		release_projectiles()
+		owner.apply_effects(weaken = 0.5)
+
+/obj/effect/xeno/shield/obj_destruction()
+	release_projectiles()
+	owner.apply_effects(weaken = 0.5)
+	return ..()
+
+///Unfeezes the projectiles on their original path
+/obj/effect/xeno/shield/proc/release_projectiles()
+	for(var/obj/projectile/proj AS in frozen_projectiles)
+		proj.flags_projectile_behavior &= ~PROJECTILE_FROZEN
+		if(istype(proj, /obj/projectile/hitscan))
+			proj.projectile_batch_move(FALSE)
+			qdel(proj)
+		else
+			START_PROCESSING(SSprojectiles, proj)
```

Comment:
I would say firing the projectile again is probably better

Reply:
Why's that?